src/test/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorDataSourceBundlerUtilsUnitTest.java

```diff
@@ -0,0 +1,99 @@
+        };
+    }
+
+    @DataProvider
+    Object[][] provideForTestGetFastaFileName() {
+        return new Object[][] {
+                { FuncotatorDataSourceBundler.OrganismKingdom.PLANTS, "actinidia_chinensis", "Actinidia_chinensis.Red5_PS1_1.69.0.cdna.all" }
+        };
+    }
+
+    @DataProvider
+    Object[][] provideForTestBuildMapWrong() {
+        return new Object[][]{
+                {
+                        FuncotatorDataSourceBundler.OrganismKingdom.PLANTS,
+                        "Aphanomyces astaci GCA_002197585.2"
+                }
+        };
+    }
+
+    @DataProvider
+    Object[][] provideForTestExtractGtfGz() {
+        return new Object[][] {
+            {
+                new File(largeFileTestDir + "funcotator/Acyrthosiphon_pisum.Acyr_2.0.51.PARTIAL.gtf.gz").getAbsolutePath(),
+                new File(largeFileTestDir + "funcotator/Acyrthosiphon_pisum.Acyr_2.0.51.PARTIAL.gtf").getAbsolutePath(),
+            },
+        };
+    }
+
+    //==================================================================================================================
+    // Tests:
+
+    @Test(dataProvider = "provideForTestGetDatasourceBaseName")
+    void testGetDatasourceBaseName(final FuncotatorDataSourceBundler.OrganismKingdom kingdom, final String speciesName, final String expectedBaseName) {
+
+        Assert.assertEquals(
+                expectedBaseName,
+                FuncotatorDataSourceBundlerUtils.getDatasourceBaseName(kingdom, speciesName)
+        );
+    }
+
+    @Test(dataProvider = "provideForTestGetFastaFileName")
+    void testGetFastaFileName(final FuncotatorDataSourceBundler.OrganismKingdom kingdom, final String speciesName, final String expectedFastaName) {
+        Assert.assertEquals(
+                FuncotatorDataSourceBundlerUtils.getFastaFileName(kingdom, speciesName),
+                expectedFastaName
+        );
+    }
+
+    @Test(dataProvider = "provideForTestBuildMapWrong", expectedExceptions = UserException.BadInput.class)
+    void testBuildMapWrong(final FuncotatorDataSourceBundler.OrganismKingdom kingdom, final String speciesName) {
+        FuncotatorDataSourceBundlerUtils.getDatasourceBaseName(kingdom, speciesName);
+    }
+
+    @Test(dataProvider = "provideForTestExtractGtfGz")
+    void testExtractGtfGz(final String gtfGzFilePath, final String expectedFilePath) {
+
+        final File destFile = getSafeNonExistentFile("testExtractGtfGz");
+        FuncotatorDataSourceBundlerUtils.extractGzFile(gtfGzFilePath, destFile.getAbsolutePath(), true);
+
+        try {
+            IntegrationTestSpec.assertEqualTextFiles(destFile, new File(expectedFilePath), "#", false);
+        }
+        catch (final IOException ex) {
+            throw new GATKException("Error: IOException encountered during test execution!", ex);
+        }
+    }
+
+    //To do: write unit tests for testing with an invalid organism name and for testing with an invalid species name
```

Comment:
Do you want to address this TODO now, or open a ticket?

Reply:
Opened ticket #7578